code/game/gamemodes/clock_cult/clock_ratvar.dm

```diff
@@ -94,8 +94,95 @@
 	icon_state = "ratvar"
 	pixel_x = -175
 	pixel_y = -175
+	var/mob/living/prey //Whoever Ratvar is chasing
+	var/clashing = FALSE //If Ratvar is FUCKING FIGHTING WITH NAR-SIE
 
 /obj/structure/clockwork/massive/ratvar/New()
 	..()
-	world << "<span class='brass'><font size=15><b>RATVAR HAS RISEN</b></font></span>"
+	SSobj.processing += src
+	flick("ratvar_spawn_anim", src)
+	world << "<span class='heavy_brass'><font size=7>\"FREE AT LAST!\"</font></span>"
 	ratvar_awakens = TRUE
+
+/obj/structure/clockwork/massive/ratvar/Destroy()
+	SSobj.processing -= src
+	world << "<span class='heavy_brass'><font size=7>\"NO! I will not... be...</font> <font size=6>banished...</font> <font size=5>again...\"</font></span>"
+	ratvar_awakens = FALSE
+	..()
+
+/obj/structure/clockwork/massive/ratvar/process()
+	for(var/atom/A in range(7, src))
+		A.ratvar_act()
+	var/dir_to_step_in = pick(cardinal)
+	if(!prey)
+		var/list/meals = list()
+		for(var/mob/living/L in living_mob_list)
+			if(L.z == z && !is_clockwork_cultist(L) && L.mind)
+				meals += L
+		if(meals.len)
+			prey = pick(meals)
+			prey << "<span class='heavy_brass'><font size=5>\"You will do.\"</font></span>\n\
+			<span class='userdanger'>Something very large and very malevolent begins lumbering its way towards you...</span>"
+	else
+		if(prob(10) || prey.stat == DEAD || is_clockwork_cultist(prey) || prey.z != z)
+			prey << "<span class='heavy_brass'><font size=5>\"How dull. Leave me.\"</font></span>\n\
+			<span class='userdanger'>You feel tremendous relief as a set of horrible eyes loses sight of you...</span>"
+			prey = null
+		else
+			if(prob(75))
+				dir_to_step_in = get_dir(src, prey)
+	loc = get_step(src, dir_to_step_in)
+
+/obj/structure/clockwork/massive/ratvar/narsie_act()
+	if(clashing)
+		return 0
+	clashing = TRUE
+	world << "<span class='heavy_brass'><font size=5>\"[pick("BLOOD GOD!!!", "NAR-SIE!!!", "AT LAST, YOUR TIME HAS COME!")]\"</font></span>"
+	world << "<span class='cult'><font size=5>\"<b>[pick("You're looking well... Ratvar.", "Ratvar! Here?! How?!", "No! Your time is over! Begone!")]</b>\"</font></span>"
+	for(var/obj/singularity/narsie/N in range(7, src))
+		clash_of_the_titans(N) //IT'S TIME FOR THE BATTLE OF THE AGES
+		break
+	return 1
+
+/obj/structure/clockwork/massive/ratvar/proc/clash_of_the_titans(obj/singularity/narsie/narsie) //IT'S TIME FOR A BATTLE OF THE ELDER GODS
+	var/winner = "Undeclared"
+	while(TRUE)
+		world << 'sound/magic/clockwork/ratvar_attack.ogg'
+		sleep(5.2)
+		for(var/mob/M in mob_list)
+			if(M.client)
+				M.client.color = rgb(75, 50, 0)
+				spawn(1)
+					M.client.color = rgb(150, 100, 0)
+					spawn(1)
+						M.client.color = initial(M.client.color)
+			shake_camera(M, 4, 3)
+		if(prob(10))
+			winner = "Ratvar"
+			break
+		sleep(rand(2,5))
+		world << 'sound/magic/clockwork/narsie_attack.ogg'
+		sleep(7.4)
+		for(var/mob/M in mob_list)
```

Comment:
@PJB3005 please dont' come onto our repo and recommend people to violate our style guidelines, pointless or not